### PR TITLE
Fix null event crash in TapGestureRecognizer

### LINKED ISSUE #43310: PointerDownEvent is null on handleTapDown
The PointerDownEvent is sometimes coming as null when chips are tapped:
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/gestures/tap.dart#L452-L459

Internal bug is b/143126092.

This might be client's own Chip implementation (not our own Material implementation). Repro instructions can be found in the internal bug.

### PR DESCRIPTION
**Pull request by dkwingsmt**
## Description

This PR fixes https://github.com/flutter/flutter/issues/43310.

The issue is caused by the following event sequence: _pointer 1 down, pointer 2 down, pointer 1 up, (timeout), pointer 2 up_. The cause is that the `PrimaryPointerRecognizer` resets the timer during `didStopTrackingLastPointer`, while states of `TapGestureRecognizer` is reset based on the events of the primary pointer, therefore the involvement of non-primary pointers can cause the timer to be triggered after `TapGestureRecognizer`'s states have been reset.

This PR fixes the issue by letting `PrimaryPointerGestureRecognizer` stop the timer when the gesture is accepted.

## Related Issues

- Fixes https://github.com/flutter/flutter/issues/43310

## Tests

I added the following tests:

- Regression test: "non-primary pointers does not trigger timtout"

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/test/gestures/tap_test.dart [hunk 1]
```diff
@@ -582,6 +582,51 @@ void main() {
     drag.dispose();
   });
 
+  testGesture('non-primary pointers does not trigger timtout', (GestureTester tester) {
```
Comment (gspencergoog) [inline packages/flutter/test/gestures/tap_test.dart]:
```suggestion
  testGesture('non-primary pointers does not trigger timeout', (GestureTester tester) {
```

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Review by goderbauer — COMMENTED** [review]
I am wondering if we should just stop the timer in the base class when the gesture gets accepted in acceptGesture. At that point there's no reason to wait for the deadline anymore. (And this would avoid leaking the timer concept from the base class to its sub classes).

**Comment by dkwingsmt** [comment]
@goderbauer Thanks for the suggestion! I've changed it to your way and everything works. Can you take another look?

File: packages/flutter/lib/src/gestures/recognizer.dart [hunk 2]
```diff
@@ -495,6 +495,7 @@ abstract class PrimaryPointerGestureRecognizer extends OneSequenceGestureRecogni
 
   @override
   void acceptGesture(int pointer) {
+    _stopTimer();
```
Comment (goderbauer) [inline packages/flutter/lib/src/gestures/recognizer.dart]: Can you also update the documentation on deadline to indicate when didExceedDeadline will not fire?

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/gestures/recognizer.dart (hunk 2 above)
Comment (dkwingsmt) [reply packages/flutter/lib/src/gestures/recognizer.dart]: Done!

**Pull request merged**